inst/include/interface/rcpp/rcpp_interface.hpp

```diff
@@ -0,0 +1,512 @@
+ * duties as a United States Government employee and thus cannot be copyrighted.
+ * This software is freely available to the public for use. The National Oceanic
+ * And Atmospheric Administration and the U.S. Government have not placed any
+ * restriction on its use or reproduction.  Although all reasonable efforts have
+ * been taken to ensure the accuracy and reliability of the software and data,
+ * the National Oceanic And Atmospheric Administration and the U.S. Government
+ * do not and cannot warrant the performance or results that may be obtained by
+ * using this  software or data. The National Oceanic And Atmospheric
+ * Administration and the U.S. Government disclaim all warranties, express or
+ * implied, including warranties of performance, merchantability or fitness
+ * for any particular purpose.
+ *
+ * Please cite the author(s) in any work or product based on this material.
+ *
+ */
+
+
+#include <vector>
+#include <map>
+
+#include "interface.hpp"
+#include "../common/model.hpp"
+#include "../common/information.hpp"
+
+
+#define RCPP_NO_SUGAR
+#include <Rcpp.h>
+
+/**
+ * @brief RcppInterface class that defines 
+ * the interface between R and C++ for parameter types.
+ */
+class parameter {
+public:
+    double value;
+    double min = std::numeric_limits<double>::min();
+    double max = std::numeric_limits<double>::max();
+    bool is_random_effect = false;
+    bool estimated = false;
+
+    parameter(double value, double min,
+            double max, bool estimated) :
+    value(value), min(min), max(max),
+    estimated(estimated) {
+
+    }
+
+    parameter(double value) {
+        this->value = value;
+    }
+
+    parameter() {
+        this->value = 0;
+    }
+};
+
+/**
+ *@brief Base class for all interface objects
+ */
+class fims_rcpp_interface_base {
+public:
+    static std::vector<fims_rcpp_interface_base*> fims_interface_objects;
+
+    virtual bool add_to_fims_tmb() {
+        std::cout << "fims_rcpp_interface_base::add_to_fims_tmb(): Not yet implemented.\n";
+        return false;
+    }
+};
+std::vector<fims_rcpp_interface_base*> fims_rcpp_interface_base::fims_interface_objects;
+
+/****************************************************************
+ * Recruitment Rcpp interface                                   *
+ ***************************************************************/
+class recruitment_interface_base : public fims_rcpp_interface_base {
+public:
+    static uint32_t id_g;
+    static std::map<uint32_t, recruitment_interface_base*> recruitment_objects;
+
+    recruitment_interface_base() {
+
+    }
+
+    virtual ~recruitment_interface_base() {
+    }
+
+};
+
+uint32_t recruitment_interface_base::id_g = 1;
+std::map<uint32_t, recruitment_interface_base* > recruitment_interface_base::recruitment_objects;
+
+/**
+ * @brief Interface class for Beverton-Holt recruitment. 
+ */
+class beverton_holt : public recruitment_interface_base {
```

Comment:
The FIMS class is SRBevertonHolt so in additional to camel casing I would call this something like SRBevertonHoltInterface